pkg/kubelet/cri/streaming/portforward/httpstream.go

```diff
@@ -252,11 +258,26 @@ func (h *httpStreamHandler) portForward(p *httpStreamPair) {
 	err := h.forwarder.PortForward(ctx, h.pod, h.uid, int32(port), p.dataStream)
 	klog.V(5).InfoS("Connection request done invoking forwarder.PortForward for port", "connection", h.conn, "request", p.requestID, "port", portString)
 
-	if err != nil {
-		msg := fmt.Errorf("error forwarding port %d to pod %s, uid %v: %v", port, h.pod, h.uid, err)
-		utilruntime.HandleError(msg)
-		fmt.Fprint(p.errorStream, msg.Error())
+	// happy path, we have successfully completed forwarding task
+	if err == nil {
+		return
 	}
+
+	// forwarding in one direction has been completed.
+	// other direction: we got RST, so we reset the data && err stream
+	if errors.Is(err, syscall.EPIPE) {
+		klog.V(5).InfoS("done invoking forwarder.PortForward but got EPIPE", "conn", h.conn, "request", p.requestID, "port", portString)
+		p.dataStream.Reset()
+		p.errorStream.Reset()
+		streamHasBeenReset = true
```

Comment:
Would it make sense to wrap it into a named func and use it in both places - here and when you defer it?

Reply:
May be I'm missing something, but wouldn't defer work as expected if  you leave only `return` here? Why to call `Reset` explicitly?

Reply:
Thanks for the timely review --- I've fixed it, PTAL thx.

Reply:
After looking at this again I'm wondering would it be clearer without defer and an additional function? After all we'll just need 2 calls of `Close` at the end of this function and two calls of 'Reset' here.

Reply:
I think defer is needed. In addition to the end of the code block execution, it will also be executed after panic occurs in `h.forwarder.PortForward`.